system.h

```diff
@@ -87,13 +88,91 @@ class System {
         api.add_api_variable("heap_free", new const APICallbackUint32(get_heap_free));
         api.add_api_variable("heap_used", new const APICallbackUint32(get_heap_used));
 
+    
+        actuator_.main_loop_.set_t0();
+        float t0 = actuator_.main_loop_.get_t0();
+        float starting_position = 0;
+        
+        MotorMode mode_ = POSITION;
+        
+        MotorCommand cmd = {};
+        cmd.mode_desired = POSITION;
+        cmd.position_desired = actuator_.main_loop_.status_.motor_position;
+        actuator_.main_loop_.set_command(cmd);
+        
+        float init_pos_enabled = 1; //actuator_.main_loop_.param_.system_controller.init_pos_enabled;
+        float init_pos = 1; //actuator_.main_loop_.param_.system_controller.init_pos;
+
+        float micro_on = 1;
+
+        float pos = actuator_.main_loop_.status_.motor_position;
+        
+        while(init_pos_enabled) {
+            if(actuator_.main_loop_.status_.motor_position > init_pos) {
+                pos += 0.0001;
+            } else if(actuator_.main_loop_.status_.motor_position < init_pos) {
+                pos -= 0.0001;
+            } else {
+                init_pos_enabled = 0;
+            }
+            
+            MotorCommand cmd = {};
+            cmd.mode_desired = POSITION;
+            cmd.position_desired = pos;
+            actuator_.main_loop_.set_command(cmd);
+        }
+        
         while(1) {
             count_++;
+            double t = count_/10000;
```

Comment:
this isn't running at fixed frequency.  You could either do a wait or connect this to an interrupt. For simplicity a wait may be a good way to start.
```
uint32_t tnext = get_clock()
while(1) {
  % do stuff
  tnext += CPU_FREQUENCY/1000  % 1 KHz
  while((int32_t) (tnext-get_clock()) < 0); % wait
}
```